### DISCUSSION TITLE
Configurable `NEXT_PUBLIC_` prefix

### DISCUSSION THREAD

**Discussion by igo** [vercel/next.js#66458] [Ideas]
### Goals

Make prefix `NEXT_PUBLIC_` configurable

### Background

Making software that is built on top of Next.js requires configuring variables prefixed with `NEXT_PUBLIC_`. This is confusing for end user who are not familiar with Next.js. People are wondering that is "NEXT". It also expose implementation details which is not a good practice

### Proposal

Make prefix configurable so I can rename it to for example `MYAPP_PUBLIC_`

**Comment by serdar-webdev** [comment]
Good idea, well done!

**Comment by Netail** [comment]
Made a draft PR for this idea: https://github.com/vercel/next.js/pull/67791 :)
